test/strategy-test-case.ts

```diff
@@ -0,0 +1,45 @@
+export interface IStrategyTestCase {
```

Comment:
We prob want to avoid the `I` on the `interfaces type`  name, is redundant and the editor already tells you that is an `Interface`,  instead I think is best to make sure the name represent the intention of the type. For example, if we have a strategy (`interface`) that a can be tested, its the name could be `TestableStrategy`  it would make much more sense as the name of the interface is more clear in what it is and what it represents.